core/src/main/java/org/apache/druid/timeline/SegmentId.java

```diff
@@ -266,6 +266,7 @@ public static SegmentId dummy(String dataSource, int partitionNum)
   private final long intervalEndMillis;
   @Nullable
   private final Chronology intervalChronology;
+  private final Interval interval;
```

Comment:
The comment above mentions that `interval` is not stored, on purpose, in order to reduce heap usage of SegmentId. Presumably that refers to heap usage on Coordinators and Brokers for clusters that have a large number of segments.

Is it important for performance that the interval be cached?

Reply:
https://implydata.atlassian.net/browse/IMPLY-17966 indicates that creating an Interval each time leads to a significant overhead.

Reply:
The impact when using the current approach is insignificant, and this change can be avoided